vsintegration/src/FSharp.Editor/Common/Extensions.fs

```diff
@@ -40,6 +42,45 @@ type Document with
                 languageServices.GetService<'T>()
                 |> Some
 
+type TextLine with
+
+    member this.ToFSharpTextSpan() = TextSpan(this.Span.Start, this.Span.End)
+
+type TextLineCollection with
+
+    member this.ToFSharpTextLineCollection() =
+        { new ITextLineCollection with
+
+            member __.Item with get index = this.[index].ToFSharpTextSpan()
+
+            member __.Count = this.Count
+        }
+
+
+type SourceText with
+
+    member this.ToFSharpSourceText() =
```

Comment:
Consider: caching these with a `ConditionalWeakTable` so that you don't end up creating a bunch of garbage for sequential calls from VS into Roslyn.

Reply:
I added it in. I was using it incorrectly too and should be resolved when I used `GetValue`